src/AeroSharp/DataAccess/MapAccess/IMap.cs

```diff
@@ -19,28 +20,41 @@ public interface IMap<TKey, TValue>
     /// <param name="value"> The value to add. </param>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
     /// <returns> A task that represents the asynchronous operation of adding the entry. </returns>
-    Task PutAsync(TKey key, TValue value, CancellationToken cancellationToken);
+    Task PutAsync(TKey key, TValue value, CancellationToken cancellationToken, params CTX[] context);
 
     /// <summary>
     ///     Asynchronously retrieve the map entry associated with the given key.
     /// </summary>
     /// <param name="key"> The key. </param>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
+    /// <param name="context"> The context within the map to apply the function, default the top-level of map. </param>
     /// <returns> A <see cref="Task{TResult}" /> which will complete with the value associated with the given key. </returns>
-    Task<KeyValuePair<TKey, TValue>> GetByKeyAsync(TKey key, CancellationToken cancellationToken);
+    Task<KeyValuePair<TKey, TValue>> GetByKeyAsync(TKey key, CancellationToken cancellationToken, params CTX[] context);
 
     /// <summary>
     ///     Asynchronously remove the map entry associated with the given key and return it.
     /// </summary>
     /// <param name="key"> The key. </param>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
+    /// <param name="context"> The context within the map to apply the function, default the top-level of map. </param>
     /// <returns> A <see cref="Task{TResult}" /> which will complete with the value associated with the given key. </returns>
-    Task<KeyValuePair<TKey, TValue>> RemoveByKeyAsync(TKey key, CancellationToken cancellationToken);
+    Task<KeyValuePair<TKey, TValue>> RemoveByKeyAsync(TKey key, CancellationToken cancellationToken, params CTX[] context);
 
     /// <summary>
     ///     Asynchronously delete the map.
     /// </summary>
     /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
     /// <returns> A task that represents the asynchronous operation of deleting the map. </returns>
     Task DeleteAsync(CancellationToken cancellationToken);
-}
+
+    /// <summary>
+    ///    Asynchronously Get {key: value} entry where map.value is the ith smallest value where i == rank, starting
+    ///    at zero being the lowest value.
+    /// </summary>
+    /// <param name="rank"> The rank to fetch from the map. </param>
+    /// <param name="context"> The context within the map to apply the function, defaults to the top-level of map.   </param>
+    /// <param name="cancellationToken"> A cancellation token to cooperatively cancel the operation. </param>
```

Comment:
These don't match the order of the method.

Reply:
Bah I thought I caught em all!